src/universal-dapp.js

```diff
@@ -556,15 +557,7 @@ UniversalDApp.prototype.linkBytecode = function (contractName, cb) {
     if (err) {
       return cb(err);
     }
-    var libLabel = '__' + libraryName + Array(39 - libraryName.length).join('_');
-    var hexAddress = address.toString('hex');
-    if (hexAddress.slice(0, 2) === '0x') {
-      hexAddress = hexAddress.slice(2);
-    }
-    hexAddress = Array(40 - hexAddress.length + 1).join('0') + hexAddress;
-    while (bytecode.indexOf(libLabel) >= 0) {
-      bytecode = bytecode.replace(libLabel, hexAddress);
-    }
+    bytecode = solc.linkBytecode(bytecode, { [libraryName]: ethJSUtil.addHexPrefix(address.toString('hex')) });
```

Comment:
`libraryName` is a literal, while `[libraryName]` will use the referenced value as key